File: Free SysLog/Support Code/My Custom ListViews.cs
using System;
using System.Windows.Forms;

namespace Free_SysLog
{
    // This class extends the ListViewItem so that I can add more properties to it for my purposes.
    public class MyReplacementsListViewItem : ListViewItem, ICloneable
    {
        public bool BoolRegex { get; set; }
        public bool BoolCaseSensitive { get; set; }
        public bool BoolEnabled { get; set; }

        public MyReplacementsListViewItem(string strInput)
        {
            Text = strInput;
        }
    }

    // This class extends the ListViewItem so that I can add more properties to it for my purposes.
    public class MyIgnoredListViewItem : ListViewItem, ICloneable
    {
        public bool BoolRegex { get; set; }
        public bool BoolCaseSensitive { get; set; }
        public bool BoolEnabled { get; set; }

        public MyIgnoredListViewItem(string strInput)
        {
            Text = strInput;
        }
    }

    public class AlertsListViewItem : ListViewItem, ICloneable
    {
        public bool BoolRegex { get; set; }
        public bool BoolCaseSensitive { get; set; }
        public string StrLogText { get; set; }
        public string StrAlertText { get; set; }
        public AlertType AlertType { get; set; }
        public bool BoolEnabled { get; set; }
        public bool BoolLimited { get; set; }

        public AlertsListViewItem(string strInput)
        {
            Text = strInput;
        }
    }

    public class ServerListViewItem : ListViewItem, ICloneable
    {
        public bool BoolEnabled { get; set; }
        public string StrName { get; set; } = null;

        public ServerListViewItem(string strInput)
        {
            Text = strInput;
        }
    }
}

File: Free SysLog/Support Code/My Custom ListViews.vb
Public Enum IgnoreType
    MainLog = 0
    RemoteApp = 1
End Enum

' This class extends the ListViewItem so that I can add more properties to it for my purposes.
Public Class MyReplacementsListViewItem
    Inherits ListViewItem
    Implements ICloneable
    Public Property BoolRegex As Boolean
    Public Property BoolCaseSensitive As Boolean
    Public Property BoolEnabled As Boolean

    Public Sub New(strInput As String)
        Me.Text = strInput
    End Sub
End Class

' This class extends the ListViewItem so that I can add more properties to it for my purposes.
Public Class MyIgnoredListViewItem
    Inherits ListViewItem
    Implements ICloneable
    Public Property BoolRegex As Boolean
    Public Property BoolCaseSensitive As Boolean
    Public Property BoolEnabled As Boolean
    Public Property IgnoreType As IgnoreType
    Public Property timeSpanOfLastOccurrence As TimeSpan
    Public Property dateOfLastOccurrence As Date
    Public Property intHits As Integer
    Public dateCreated As Date
    Public strComment As String

    Public Sub New(strInput As String)
        Me.Text = strInput
    End Sub
End Class

Public Class AlertsListViewItem
    Inherits ListViewItem
    Implements ICloneable
    Public Property BoolRegex As Boolean
    Public Property BoolCaseSensitive As Boolean
    Public Property StrLogText As String
    Public Property StrAlertText As String
    Public Property AlertType As AlertType
    Public Property BoolEnabled As Boolean
    Public Property BoolLimited As Boolean

    Public Sub New(strInput As String)
        Me.Text = strInput
    End Sub
End Class

Public Class ServerListViewItem
    Inherits ListViewItem
    Implements ICloneable
    Public Property BoolEnabled As Boolean
    Public Property StrName As String = Nothing

    Public Sub New(strInput As String)
        Me.Text = strInput
    End Sub
End Class